### Projet 1 
Groupe 3

Mathieu Muty - MUTM22090200

Gabrielle Walgraef - WALG13580100

Mamoudou Camara - CAMM12080206

In [1]:
# importation des libraries
import numpy as np
import pandas as pd

In [3]:
# chargement des fichiers csv
environnement = pd.read_csv('./data_projet/environnement.csv')
genre = pd.read_csv('./data_projet/genre.csv')
meteo = pd.read_csv('./data_projet/meteo.csv')
propulation = pd.read_csv('./data_projet/population.csv')
surface = pd.read_csv('./data_projet/surface.csv')
rapport = pd.read_excel('./data_projet/rapports-accident-2021.xlsx')

## Question 1
Les fichiers suivants : environnement, genre, meteo, population ainsi que surface ne contiennent pas de valeur
aberrante puisque ces fichiers nous donne les codes pour mieux comprendre le fichier excel 'rapport-accident-2021'

Nous nous concentrons donc sur le fichier 'rapports-accident-2021.xlsx' afin de déceler les valeurs extrêmes 
ou aberrantes

In [27]:
print(rapport.isna().sum())

print("type de donnée de la colonne CD_GENRE_ACCDN :", rapport['CD_GENRE_ACCDN'].dtype)
print("type de donnée de la colonne CD_GENRE_ACCDN :", rapport['MRC'].dtype)
print("nombre de valeur du dataframe : ", len(rapport))
# CD_COND_METEO meteo -> imputer nan by autre (code 99)
# CD_ENVRN_ACCDN environnement -> imputer nan by non précisé (code 0)
# CD_GENRE_ACCDN genre -> impossible de le savoir -> fillna('Unknown')(question 5 utilise cette column)
# CD_ETAT_SURFC surface -> imputer nan by autre (code 99)
# CD_ECLRM eclairage -> impossible de le savoir (non utilisé par les questions) - 
# VITESSE_AUTOR -> impossible de le savoir (non utilisé par les questions)

DT_ACCDN                          0
HR_ACCDN                          0
GENRE_CONDUCTEUR                  0
AGE_CONDUCTEUR                    0
gravite                           0
VALEURS_DOMMAGES                  0
NB_MORTS                          0
NB_BLESSES_GRAVES                 0
NB_BLESSES_LEGERS                 0
NB_VEH_IMPLIQUES_ACCDN            0
NB_DECES_PIETON                   0
NB_BLESSES_PIETON                 0
NB_VICTIMES_PIETON                0
NB_DECES_MOTO                     0
NB_BLESSES_MOTO                   0
NB_VICTIMES_MOTO                  0
NB_DECES_VELO                     0
NB_BLESSES_VELO                   0
NB_VICTIMES_VELO                  0
REG_ADM                          15
MRC                              15
VITESSE_AUTOR                  9310
CD_GENRE_ACCDN                  798
CD_ETAT_SURFC                  1163
CD_ECLRM                       1415
CD_ENVRN_ACCDN                 1239
CD_COND_METEO                  1352
nb_automobile_camion_leger  

## Démarche par rapport aux valeurs manquantes
Après identification des colonnes utilisées dans les prochaines questions et qui contiennent des valeurs manquantes,
Nous avons remarqué que ce cas de figure se retrouve dans la question 5 et la question 6

#### Cas de la question 5 :
Dans la question 5 nous utilisons la colonne **'CD_GENRE_ACCDN'** qui contient **798** valeurs manquantes.
Cette colonne nous indique un code **numérique** qui correspond au genre de l'accident.

Après étude de la question 5, qui incorpore des pourcentages a réaliser, nous avons décidé de supprimer ces valeurs manquantes pour le traitement de cette question même si ce nombre n'est pas significatif par rapport à la taille du dataFrame.


#### Cas de la question 6:
Dans la question 6 nous utilisons la colone **MRC** qui contient **15** valeurs manquantes. 
Après étude de la question 6, nous avons décidé de les supprimer afin de créer le dataframe **agg_tx_acc** sans erreur potentielle, de plus la suppression de ces valeurs n'a aucune incidence sur la question puisque nous devons sélectionner uniquement les 5 **MRC** avec le plus haut taux d'accident global.

## Question 3

In [32]:
rapport['Date'] = pd.to_datetime(rapport['DT_ACCDN'], format='%d/%m/%Y')

EstWeekEnd = []

for date in rapport['Date']:
    estWeekEnd = date.weekday() >= 5
    EstWeekEnd.append(estWeekEnd)
    
print(EstWeekEnd)

[True, False, True, False, False, False, True, True, False, True, False, False, False, False, False, False, False, False, False, False, False, True, True, False, False, False, True, True, True, False, False, False, False, False, False, True, True, False, False, False, False, False, False, True, True, False, False, False, False, False, False, False, False, True, True, False, False, False, False, False, False, False, False, False, False, True, False, False, False, False, False, False, False, False, False, True, False, False, False, False, False, False, True, False, False, True, True, False, False, False, False, False, False, False, False, False, False, False, False, False, False, False, False, True, False, False, False, False, False, False, False, True, False, False, False, False, False, False, False, False, False, True, True, False, False, True, False, False, True, True, True, False, False, False, False, True, False, False, False, True, True, True, True, True, True, False, False, False,

In [26]:
# a appliquer pour la question 5 : 
mask = rapport['CD_GENRE_ACCDN'].isna()
clean_report = rapport.drop(rapport[mask].index, axis=0)
print(clean_report.isna().sum())

DT_ACCDN                          0
HR_ACCDN                          0
GENRE_CONDUCTEUR                  0
AGE_CONDUCTEUR                    0
gravite                           0
VALEURS_DOMMAGES                  0
NB_MORTS                          0
NB_BLESSES_GRAVES                 0
NB_BLESSES_LEGERS                 0
NB_VEH_IMPLIQUES_ACCDN            0
NB_DECES_PIETON                   0
NB_BLESSES_PIETON                 0
NB_VICTIMES_PIETON                0
NB_DECES_MOTO                     0
NB_BLESSES_MOTO                   0
NB_VICTIMES_MOTO                  0
NB_DECES_VELO                     0
NB_BLESSES_VELO                   0
NB_VICTIMES_VELO                  0
REG_ADM                           7
MRC                               7
VITESSE_AUTOR                  9164
CD_GENRE_ACCDN                    0
CD_ETAT_SURFC                  1047
CD_ECLRM                       1303
CD_ENVRN_ACCDN                 1127
CD_COND_METEO                  1223
nb_automobile_camion_leger  

In [29]:
# a appliquer pour la question 6
mask = rapport['MRC'].isna()
clean_report = rapport.drop(rapport[mask].index, axis=0)
print(clean_report.isna().sum())

DT_ACCDN                          0
HR_ACCDN                          0
GENRE_CONDUCTEUR                  0
AGE_CONDUCTEUR                    0
gravite                           0
VALEURS_DOMMAGES                  0
NB_MORTS                          0
NB_BLESSES_GRAVES                 0
NB_BLESSES_LEGERS                 0
NB_VEH_IMPLIQUES_ACCDN            0
NB_DECES_PIETON                   0
NB_BLESSES_PIETON                 0
NB_VICTIMES_PIETON                0
NB_DECES_MOTO                     0
NB_BLESSES_MOTO                   0
NB_VICTIMES_MOTO                  0
NB_DECES_VELO                     0
NB_BLESSES_VELO                   0
NB_VICTIMES_VELO                  0
REG_ADM                           0
MRC                               0
VITESSE_AUTOR                  9302
CD_GENRE_ACCDN                  790
CD_ETAT_SURFC                  1154
CD_ECLRM                       1406
CD_ENVRN_ACCDN                 1231
CD_COND_METEO                  1343
nb_automobile_camion_leger  